File: Python/Challenge/September/03_Sum_of_Digits_of_String_After_Convert.py
# https://leetcode.com/problems/sum-of-digits-of-string-after-convert/description/?envType=daily-question&envId=2024-09-03

def getLucky(s: str, k: int) -> int:
    # aa = ''
    # for i in s:
    #     aa += str(ord(i) - ord('a') + 1)
    # # 88
    # result = 0
    # for _ in range(k):
    #     new_sum = 0
    #     for i in aa:
    #         new_sum += int(i)
    #     aa = str(new_sum)
    # return aa

    num_str = ''.join(str(ord(char) - ord('a') + 1) for char in s)
    for _ in range(k):
        num_str = str(sum(int(digit) for digit in num_str))
    return int(num_str)


s = "iiii"
k = 2
print(getLucky(s, k))

File: PHP/Challenge/September/03_Sum_of_Digits_of_String_After_Convert.php
<?php
// https://leetcode.com/problems/sum-of-digits-of-string-after-convert/description/?envType=daily-question&envId=2024-09-03
function getLucky($s, $k) {
    $num_str = "";
    $length = strlen($s);

    for ($i = 0; $i < $length; $i++) {
        $char = $s[$i];
        $char_value = ord($char) - ord('a') + 1;
        $num_str .= $char_value;
    }

    for ($j = 0; $j < $k; $j++) {
        $new_sum = 0;
        $length = strlen($num_str);

        for ($i = 0; $i < $length; $i++) {
            $new_sum += intval($num_str[$i]);
        }

        $num_str = strval($new_sum);
    }

    return intval($num_str);
}

$s = "zbax";
$k = 2;
echo getLucky($s, $k);